### PR TITLE
[iOS, Android] Upgrade RN to 0.59.2

### PR DESCRIPTION
**Pull request by rasom**
status: ready

### CODE REVIEW

**Comment by jeluard** [comment]
RN 0.59 has just been released :sweat_smile:

**Comment by rasom** [comment]
@jeluard well if it's just released it might still contain some regressions which will require updates in some deps, etc... So probably we will need to finish this step first anyway

**Review by mandrigin — APPROVED** [review]
Gradle, pods and iOS updates look good

File: android/build.gradle [hunk 1]
```diff
@@ -1,21 +1,56 @@
...
+        classpath 'com.google.gms:google-services:4.0.1'
         classpath 'de.undercouch:gradle-download-task:3.1.2'
 
         // NOTE: Do not place your application dependencies here; they belong
         // in the individual module build.gradle files
     }
 }
 
+subprojects {
+    /*project.configurations.all {
```
Comment (jeluard) [inline android/build.gradle]: Remove?

File: android/build.gradle [hunk 2]
```diff
@@ -1,21 +1,56 @@
...
+        resolutionStrategy.eachDependency { details ->
+            if (details.requested.group == 'com.android.support'
+                    && !details.requested.name.contains('multidex') ) {
+                details.useVersion "27.1.1"
+            }
+        }
+    }*/
+    afterEvaluate {
+        if (project.hasProperty("android") && true
+                /*(project.name == 'react-native-config'
```
Comment (jeluard) [inline android/build.gradle]: Remove?

File: android/build.gradle [hunk 3]
```diff
@@ -1,21 +1,56 @@
...
     repositories {
         flatDir { dirs "libs", "${rootDir}/app/libs" }
         google()
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.0.1'
-        classpath 'com.google.gms:google-services:4.1.0'
+        classpath 'com.android.tools.build:gradle:3.3.2'
+        classpath 'com.google.gms:google-services:4.0.1'
```
Comment (jeluard) [inline android/build.gradle:20]: Downgraded?

File: android/app/build.gradle [hunk 4]
```diff
@@ -266,15 +274,10 @@ dependencies {
 
     compile 'com.github.status-im:function:0.0.1'
     implementation fileTree(dir: "node_modules/realm/android/libs", include: ["*.jar"])
-    compile 'com.facebook.fresco:animated-gif:1.9.0'
 }
 
 // Run this once to be able to run the application with BUCK
 // puts all compile dependencies into folder libs for BUCK to use
```
Comment (jeluard) [inline android/app/build.gradle:279]: Remove?

File: android/app/build.gradle [hunk 5]
```diff
@@ -238,10 +245,11 @@ dependencies {
     implementation project(':react-native-dialogs')
     implementation project(':realm')
     implementation fileTree(dir: "libs", include: ["*.jar"])
-    implementation "com.android.support:appcompat-v7:28.0.0"
+    implementation "com.android.support:appcompat-v7:28.+"
     // Force using exact RN version instead of relying on gradle dependency resolution
```
Comment (jeluard) [inline android/app/build.gradle]: Remove?

File: android/gradle.properties [hunk 6]
```diff
@@ -17,15 +17,14 @@
 # http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
 # org.gradle.parallel=true
 
-android.useDeprecatedNdk=true
+# android.useDeprecatedNdk=true
```
Comment (jeluard) [inline android/gradle.properties]: Remove?

File: mobile_files/package.json.orig [hunk 7]
```diff
@@ -70,6 +72,6 @@
     "web3-utils": "1.0.0-beta.36"
   },
   "devDependencies": {
-    "@babel/core": "7.0.0-beta.47"
+    "@babel/core": "^7.3.4"
```
Comment (jeluard) [inline mobile_files/package.json.orig]: Needed twice?

File: src/status_im/react_native/resources.cljs [hunk 8]
```diff
@@ -69,7 +69,8 @@
    :hardwallet-card     (js/require "./resources/images/ui/hardwallet-card.png")
    :secret-keys         (js/require "./resources/images/ui/secret-keys.png")
    :keycard-lock        (js/require "./resources/images/ui/keycard-lock.png")
-   :hold-card-animation (js/require "./resources/images/ui/hold-card-animation.png")
+   :hold-card-animation (try (js/require "./resources/images/ui/hold-card-animation.gif")
```
Comment (jeluard) [inline src/status_im/react_native/resources.cljs:72]: Why?

File: android/gradle.properties [hunk 9]
```diff
@@ -17,15 +17,14 @@
...
 # To create a new keystore for development purposes (https://developer.android.com/studio/publish/app-signing.html)
 # $keytool -genkey -v -keystore ./status-keystore.jks -keyalg RSA -keysize 2048 -validity 10000 -alias status
 STATUS_RELEASE_STORE_FILE=~/.gradle/status-im.keystore
 STATUS_RELEASE_STORE_PASSWORD=password
 STATUS_RELEASE_KEY_ALIAS=status
 STATUS_RELEASE_KEY_PASSWORD=password
 
-# Workaround for issue https://github.com/facebook/react-native/issues/16906 (TODO: fixed in 0.57.3, remove once we upgrade react-native)
 android.enableAapt2=false
```
Comment (jakubgs) [inline android/gradle.properties]: Wait, shouldn't this be removed based on the comment? Or am I missing something?

File: android/app/build.gradle (hunk 4 above)
Comment (rasom) [reply android/app/build.gradle:279]: nope, it seems it was moved few lines above, will move back

File: src/status_im/react_native/resources.cljs (hunk 8 above)
Comment (rasom) [reply src/status_im/react_native/resources.cljs:72]: because `learn test-cljs` fails as `.gif` file can't be required. Why? Have no idea. 
That file originally was named as `something.png` which is definitely not the case, and it worked fine but after upgrade android refuses to put invalid png into resources.

File: android/gradle.properties (hunk 9 above)
Comment (rasom) [reply android/gradle.properties]: I'll try that separately. I suppose there is a lot of other garbage in this PR, because in order to make it work I had to come with many senseless suggestions and try them, and some are still here even if have no effect ¯\_(ツ)_/¯

Comment (jakubgs) [reply android/gradle.properties]: I see, thanks!

File: android/build.gradle (hunk 3 above)
Comment (rasom) [reply android/build.gradle]: fixed

File: android/build.gradle (hunk 2 above)
Comment (rasom) [reply android/build.gradle]: fixed

File: android/build.gradle (hunk 1 above)
Comment (rasom) [reply android/build.gradle]: fixed

File: android/app/build.gradle (hunk 4 above)
Comment (rasom) [reply android/app/build.gradle:279]: fixed

File: android/gradle.properties (hunk 6 above)
Comment (rasom) [reply android/gradle.properties]: fixed

File: android/gradle.properties (hunk 9 above)
Comment (rasom) [reply android/gradle.properties]: app can be built without `android.enableAapt2=false`, removed

File: android/app/build.gradle (hunk 5 above)
Comment (rasom) [reply android/app/build.gradle]: it hasn't been touched in this PR so is not relevant

File: mobile_files/package.json.orig (hunk 7 above)
Comment (rasom) [reply mobile_files/package.json.orig]: fixed

**Comment by rasom** [comment]
@jeluard all comments are addressed

**Review by jakubgs — APPROVED** [review]
I can't vouch for the `react-native` changes in JavaScript but the ones in Nix, Gradle, and build scripts look good.

**Review by jeluard — APPROVED** [review]

**Comment by churik** [comment]
On Android on app start:

[logcat.log](https://github.com/status-im/status-react/files/3017122/logcat.log)

**Comment by rasom** [comment]
@churik should be fixed in the next build

File: android/build.gradle (hunk 3 above)
Comment (rasom) [reply android/build.gradle:20]: added comment to code, it actually fixes one problem https://github.com/invertase/react-native-firebase/issues/766#issuecomment-365116012

**Comment by antdanchenko** [comment]
@churik should I update tests or you are going to update them?

**Comment by churik** [comment]
@rasom 
### 1) Error when tap on "Join public chat button"
Lg_tr
On tap on "Exit"  app is crashing.

Logs: 
[Status-debug-logs (3).zip](https://github.com/status-im/status-react/files/3017762/Status-debug-logs.3.zip)
**Device:** LG V20
**OS:** Android 8

### 2) White screen and no header are shown in webview when you open DApp 
Instead of spinnier animation white screen is displayed, and  the header is shown only after you scroll view.
no_header

Video: https://monosnap.com/file/0QllNZdbQ3QsOFiUDDdJNyM0d3jwuN
**Device:** LG V20
**OS:** Android 8

### 3) Spinner has gone when buying stickers, @flexsurfer did we remove it?
image

### 4) Receive transaction screen is broken
image

**Comment by churik** [comment]
@antdanchenko going to launch nightly tests against this PR as well - probably more fixes in tests will be needed, so suggest you to hold on for now.

**Comment by rasom** [comment]
@churik 1) and 2) should be fixed in the next build

**Comment by rasom** [comment]
@churik @asemiankevich can you please add new issues separately next time, It might be tricky to follow updated comments

**Comment by annadanchenko** [comment]
@rasom 
### issue 5 long tap on message -> no popup to select reply is shown
real Pixel XL Android 9 with build 51  and real iphone 7+ iOS 12.1.4 with build 48

steps:
- join public chat
- long tap on someone's message

expected: popup is shown with Reply, Copy, Share options
actual: nothing happens

**Comment by churik** [comment]
@rasom 
sorry, one more commit for fixing e2e tests will be added  (turned out from nightly set of tests (it is much wider than set of e2e for PRs), that one more locator is changed).
Please, squash them with yours when PR will be ready to merge.
For now as I can understand fixes for https://github.com/status-im/status-react/pull/7666#issuecomment-477639926 and issue 4 should be added.

Thanks!

**Comment by flexsurfer** [comment]
@churik  3) no we don't ,it should work

**Comment by rasom** [comment]
@churik I'm not sure where I have to expect that spinner in `3)` would be great to have steps to reproduce or something. I'm trying to find it in `develop` but no results till now.

**Comment by asemiankevich** [comment]
> @churik I'm not sure where I have to expect that spinner in `3)` would be great to have steps to reproduce or something. I'm trying to find it in `develop` but no results till now.

@rasom 

it is reproducible in PR build and e2e and in develop not, beacuse stickers are hidden in develop. Steps:

1. open status
2. switch to ropsten
3. faucet some funds
4. request some STT
5. open 1x1 chat
6. tap stickers icon at the bottom right
7. select sticker pack -> tap the button with the price and complete transaction
8. pay attention to the button with the price -> it should show spinner until transaction is being mined.
9. once mined, the price button will change to Install

**Comment by rasom** [comment]
@asemiankevich it is reproducible in `develop` locally but not in nightly. Should be fixed now, can you check please?

**Comment by churik** [comment]
restarting whole set of e2e tests in https://ci.status.im/job/end-to-end-tests/job/status-app-nightly/799/

**Comment by rasom** [comment]
`5)` is RN regression https://github.com/facebook/react-native/issues/24001

**Comment by churik** [comment]
@rasom 5 looks to me like a blocker for this PR. Is it possible to make upgrade to RN-version without commit that causes the issue or what can we do to avoid this?

**Comment by rasom** [comment]
@churik `5)` is fixed in our fork now, the next build should contain the fix

**Comment by Serhy** [comment]
6) Android 8.1 (but OK with iOS!): camera view overlaps top app bar (where `<` Back button is present). But if to tap the top-left corner on a place where `<` bar is located I back to previous screen.
Issue is present on next camera views:
`Home` -> `+` -> `Start new chat` -> `ScarQRcode`
`Home` -> `+` -> -> `ScarQRcode`
`Profile` -> `Mailserver` -> `Add mailserver` -> top-right `+` -> `ScarQRcode`
`Profile` -> `Bootnodes` -> top-right `+` -> `ScarQRcode`
Screenshot 2019-03-29 at 12 38 59

**Comment by rasom** [comment]
I see that's not going to be easy... Working on `6)`

**Comment by rasom** [comment]
@Serhy @churik `6)` should be fixed now

**Comment by churik** [comment]
@rasom 
#### 6) header still is cut 
cut

#### 7) Replies don't fit box (Android)

**Steps:**
- open any chat 
- send message > 8 rows
- long tap > Reply
sdfsfd

**Comment by rasom** [comment]
@churik is that `6)` `New contact` scanning? Guys, please, be more specific when you add these issues, it's a bit hard to follow some of them. This PR is not about some specific part of codebase and it is hard grasp what's wrong and where when only some screen is added. So that new `6)` is similar but a different issue, not the same as @Serhy posted originally.

**Comment by rasom** [comment]
@churik sorry, I probably just missed it in original comment. really need to finish this task finally :D

**Comment by churik** [comment]
@rasom 
6) sorry that I didn't mention it, it is in the same views https://github.com/status-im/status-react/pull/7666#issuecomment-477951552

**Comment by rasom** [comment]
so `7)` is again RN regression https://github.com/facebook/react-native/pull/22693

**Comment by rasom** [comment]
`7)` should be fixed

**Comment by rasom** [comment]
`6)` is finally fixed

@churik all regressions found during testing are fixed now

**Comment by churik** [comment]
**Link to tests:**
https://ethstatus.testrail.net/index.php?/runs/view/2976&group_by=cases:section_id&group_order=asc
All except [this](https://ethstatus.testrail.net/index.php?/tests/view/564671), which is fixed in https://github.com/status-im/status-react/pull/7666/commits/78425e3686cad72253f9f74b10785c5e5002a6fa all of them are failures in nightly as well or false failures.

**Tested on last builds (IOS 11.4.1, Android 8.1):**
- all 7 issues are fixed
- can add custom mailserver \ bootnode
- chatting in public\1-1 chats
- open DApps
- DApps permissions
- links \ emoji in chats
- long messages in public chats
- install \ disable extensions
- deep links to 1-1 \ public chats \ Dapps
- started chat with valid ENS name \ can send transaction to ENS name from wallet
- can reserve ENS Name with ENS DApp
- korean \ arabic text
- PNs when user logged in (IOS and Android) \ app is closed (Android only)

**Tested on previous builds:**
- transactions from wallet - send, request, transaction view
- /send, /request from 1-1 chat
- stickers - adding, sending
- restoring account with assets 
- collectibles - view, adding
- browser back \ forward
- popup when you switch to mobile network
- set custom avatar \ username
- switching mailservers \ bootnodes \ development mode \ networks \ log settings
- DApps transaction \ contract deploying \ message signing
- group chats
- device pairing
- blocking \ unblocking user

Also checked that Mac O \ Linux \ Windows builds can be installed (fresh install) and used.

@rasom please, take 4 my commits when squashing all before merge.
